pynws/forecast.py

```diff
@@ -65,67 +74,109 @@ def _parse_duration(duration_str: str) -> timedelta:
+            Generator[dict[Detail, DetailValue]]: Sequence of forecast details
+            corresponding with the list of times to retrieve.
+        """
         if not isinstance(iterable_when, Iterable):
-            raise TypeError(f"{iterable_when!r} is not an Iterable")
+            raise TypeError(f"{iterable_when!r} is not iterable")
 
         for when in iterable_when:
             yield self.get_details_for_time(when)
 
-    def get_detail_for_time(
-        self, detail: Detail, when: datetime
-    ) -> tuple[Any, str | None]:
-        """Retrieve single forecast detail for a point in time."""
+    def get_detail_for_time(self, detail: Detail, when: datetime) -> DetailValue:
+        """Retrieve single forecast detail for a point in time.
+
+        Args:
+            detail (Detail): Forecast detail to retrieve.
+            when (datetime): Point in time of requested forecast detail.
+
+        Raises:
+            TypeError: If 'detail' argument is not a 'Detail'.
+            TypeError: If 'when' argument is not a 'datetime'.
 
+        Returns:
+            DetailValue: Requested forecast detail value for the specified time.
+        """
         if not isinstance(detail, Detail):
             raise TypeError(f"{detail!r} is not a Detail")
         if not isinstance(when, datetime):
             raise TypeError(f"{when!r} is not a datetime")
 
         when = when.astimezone(timezone.utc)
-        time_values, units = self.details.get(detail)
-        if time_values and units:
-            return self._find_detail_for_time(when, time_values, units)
-        return None, None
+        time_values = self.details.get(detail)
+        return self._get_value_for_time(when, time_values) if time_values else None
 
     def get_details_by_hour(
-        self, start_time: datetime, hours: int = 12
-    ) -> Generator[dict[Detail, Any]]:
-        """Retrieve a sequence of hourly forecast details"""
+        self, start_time: datetime, hours: int = 24
+    ) -> Generator[dict[Detail, DetailValue]]:
+        """Retrieve a sequence of hourly forecast details
+
+        Args:
+            start_time (datetime): First time to retrieve.
+            hours (int, optional): Number of hours to retrieve.
+
+        Raises:
+            TypeError: If 'start_time' argument is not a 'datetime'.
 
+        Yields:
+            Generator[dict[Detail, DetailValue]]: Sequence of forecast detail
+            values with one details dictionary per requested hour.
+        """
         if not isinstance(start_time, datetime):
             raise TypeError(f"{start_time!r} is not a datetime")
 
         start_time = start_time.replace(minute=0, second=0, microsecond=0)
-        for hour in range(hours):
-            yield self.get_details_for_time(start_time + timedelta(hours=hour))
+        for _ in range(hours):
+            end_time = start_time + ONE_HOUR
+            details = {
+                Detail.START_TIME: datetime.isoformat(start_time),
```

Comment:
In the current master branch the keys are wrapped with `str`.  I'm not sure what the subtleties are here with `StrEnum` attribute as a key.

Reply:
I'm not sure why exactly, but string and enum values both work as key values. Since the other values will be `Detail` and type hints for return values and parameters are also `Detail`, I thought it best to be consistent.